### ISSUE TITLE
Azure Key Vault with DefaultAzureCredential as Secrets Manager not doable due to helm chart

### ISSUE THREAD

**Issue by Nemeczek** [open-metadata/OpenMetadata#15839]
**Affected module**
backend - helm chart

**Describe the bug**
Helm chart does not support Azure Key Vault.

**To Reproduce**
Try to use managed-azure-kv as Secret Manager:

```
openmetadata:
  config:
    secretsManager:
      enabled: true
      provider: managed-azure-kv # or env var SECRET_MANAGER. For non-managed use 'azure-kv'.
      additionalParameters:
        enabled: true
        vaultName:
          secretRef: secret
          secretKey: openmetadata-secret-manager-key-vault-name
```

**Expected behavior**
Secrets Manager based on Azure Key Vault will be working. And using DefaultAzureCredential because only KeyVault name is provided.

**Version:**
 - OpenMetadata version: 1.3.2
 - OpenMetadata Helm chart version: 1.3.2

**Additional context**

Per documentation:
![image](https://github.com/open-metadata/OpenMetadata/assets/1111404/9a0adefe-8c5a-4ad1-a220-557fb9e30035)

This only creates this:

```
apiVersion: v1
kind: Secret
metadata:
  name: openmetadata-secretsmanager-secret
type: Opaque
data:
  SECRET_MANAGER: Im1hbmFnZWQtYXp1cmUta3Yi
  SECRET_MANAGER_PREFIX: IiI=
  SECRET_MANAGER_TAGS: W10=
```

Based on the [source code of the Helm chart](https://github.com/open-metadata/openmetadata-helm-charts/blob/83b837ddc66d08ad75252f7b1ba18be665ef152c/charts/openmetadata/templates/_helpers.tpl#L311) I tried to add property enabled: true to the additionalParameters:

which actually forces chart to generate missing fields:

```
        - name: OM_SM_CLIENT_ID
          valueFrom:
            secretKeyRef:
              name: secret
              key: openmetadata-secret-manager-key-vault-name
        - name: OM_SM_CLIENT_SECRET
          valueFrom:
            secretKeyRef:
              name: secret
              key: openmetadata-secret-manager-key-vault-name
        - name: OM_SM_TENANT_ID
          valueFrom:
            secretKeyRef:
              name: secret
              key: openmetadata-secret-manager-key-vault-name
        - name: OM_SM_VAULT_NAME
          valueFrom:
            secretKeyRef:
              name: secret
              key: openmetadata-secret-manager-key-vault-name
```

Sadly it uses default values - so it is not possible to provide only KeyVaultName to use DefaultAzureCredential.
Moreover adding the enabled: true adds region to the mix:

![image](https://github.com/open-metadata/OpenMetadata/assets/1111404/d453a0f8-54a6-490d-b3ff-38d6ec40cdef)

Due to this [line of code](https://github.com/open-metadata/openmetadata-helm-charts/blob/83b837ddc66d08ad75252f7b1ba18be665ef152c/charts/openmetadata/templates/secrets.yaml#L153). Which is not needed by the Azure KV Secrets Manager. 

So I think the Helm chart is not doing what it supposed to do. If it is then the documentation should mention the importance of having enabled: true in both openmetadata.config.secretsmanager and openmetadata.config.secretsmanager.additionalParameters

I tried to have it fixed in [the chart issue](https://github.com/open-metadata/openmetadata-helm-charts/issues/222). But it was closed and @harshach suggested me to open issues here not in the chart repo.

**Comment by Nemeczek** [comment]
As per this [response there is workaround](https://github.com/open-metadata/openmetadata-helm-charts/issues/222#issuecomment-2049028645):

TLDR:
Add this to values.yaml for OM HELM chart
``` 
extraEnvs:
  - name: OM_SM_VAULT_NAME
    value: kv-stark-data-test-om
```

**Comment by pmbrull** [comment]
closed by https://github.com/open-metadata/OpenMetadata/pull/16000
